doc/snmp/snmp-configdb-migration-hld.md

```diff
@@ -0,0 +1,221 @@
+
+The new SNMP tables are: 
+SNMP
+SNMP_COMMUNITY
+SNMP_USER
+
+
+# SNMP Table
+```
+SNMP|LOCATION
+SNMP|CONTACT
+
+admin@str-s6000-acs-11:~$ redis-cli -n 4 hgetall "SNMP|LOCATION"
+1) "LOCATION"
+2) "Redmond"
+
+admin@str-s6000-acs-11:~$ redis-cli -n 4 hgetall "SNMP|CONTACT"
+1) "joe"
+2) "joe@microsoft.com"
+```
+
+# SNMP_COMMUNITY Table
+```
+SNMP_COMMUNITY|<community>
+
+admin@str-s6000-acs-11:~$ redis-cli -n 4 hgetall "SNMP_COMMUNITY|Jack"
+1) "TYPE"
+2) "RW"
+
+```
+
+# SNMP_USER Table 
+```
+SNMP_USER|<user>
+
+admin@str-s6000-acs-11:~$ redis-cli -n 4 hgetall "SNMP_USER|Travis"
+ 1) "SNMP_USER_ENCRYPTION_TYPE"
+ 2) "AES"
+ 3) "SNMP_USER_AUTH_TYPE"
+ 4) "SHA"
+ 5) "SNMP_USER_ENCRYPTION_PASSWORD"
+ 6) "TravisEncryptPass"
+ 7) "SNMP_USER_AUTH_PASSWORD"
+ 8) "TravisAuthPass"
+ 9) "SNMP_USER_TYPE"
+10) "Priv"
+11) "SNMP_USER_PERMISSION"
+12) "RO"
+```
+
+
+
+# New SNMP CLI Commands
+# Show commands
+```
+admin@str-s6000-acs-11:~$ show run snmp -h
+Usage: show run snmp [OPTIONS] COMMAND [ARGS]...
+
+  Show SNMP information
+
+Options:
+  -?, -h, --help  Show this message and exit.
+
+Commands:
+  community  show runningconfiguration snmp community
+  contact    show runningconfiguration snmp contact
+  location   show runningconfiguration snmp location
+  users      show runningconfiguration snmp users
+admin@str-s6000-acs-11:~$ 
+```
+
+show run snmp community
+```
+admin@str-s6000-acs-11:~$ show run snmp community
+Community String    Community Type
+------------------  ----------------
+Qi                  RO
+Travis              RO
+Bill                RO
+Jack                RW
+public              RO
+Joker               RW
+admin@str-s6000-acs-11:~$ 
+```
+
+show run snmp contact
+```
+admin@str-s6000-acs-11:~$ show run snmp contact
+Contact    Contact Email
+---------  -----------------
+Joe        joe@microsoft.com
+admin@str-s6000-acs-11:~$ 
+```
+
+show run snmp location
+```
+admin@str-s6000-acs-11:~$ show run snmp location 
+Location
+----------
+Redmond
+admin@str-s6000-acs-11:~$ 
+```
+
+show run snmp users
+```
+admin@str-s6000-acs-11:~$ show run snmp users
+User    Type    Auth Type    Auth Password    Encryption Type    Encryption Password
+------  ------  -----------  ---------------  -----------------  ---------------------
+Travis  Priv    SHA          TravisAuthPass   AES                TravisEncryptPass
+admin@str-s6000-acs-11:~$
+```
+
+
+# Config Commands
+```
+admin@str-s6000-acs-11:~$ sudo config snmp -h
+Usage: config snmp [OPTIONS] COMMAND [ARGS]...
+
+  SNMP configuration tasks
+
+Options:
+  -?, -h, --help  Show this message and exit.
+
+Commands:
+  community
+  contact
+  location
+  user
+admin@str-s6000-acs-11:~$ 
+```
+
+sudo config snmp community 
+```
+admin@str-s6000-acs-11:~$ sudo config snmp community -h
+Usage: config snmp community [OPTIONS] COMMAND [ARGS]...
+
+Options:
+  -?, -h, --help  Show this message and exit.
+
+Commands:
+  add  Add snmp community
+  del  Add snmp community
```

Comment:
Instead of `add` `del` individual community string, better to `set` multiple strings in one command. So there will never be a transient state, such as no string at all.

Reply:
So the none of the other networking vendor OS allow this.  All vendors only support adding and removing one at a time. I think we should follow that standard.   We should be able to have no string all if if that is what folks want.

Reply:
Suggest keep `add` `del` commands, and add a `replace` command, which is useful in your use case.

---
In reply to: [541519000](https://github.com/Azure/SONiC/pull/723#discussion_r541519000) [](ancestors = 541519000)

Reply:
Added replace command.